catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/view/Registry.view.js

```diff
@@ -261,5 +270,69 @@ define([
            }
         });
 
+        RegistryView.RegenerateSourcesModal = Marionette.ItemView.extend({
+            template: 'regenerateSourcesModal',
+            className: 'modal',
+            events: {
+                'click .submit-button' : 'regenerateSources',
+                'click .cancel-button' : 'cancel',
+                'click .close': 'cancel'
+            },
+            regenerateUrl: '/admin/jolokia/exec/org.codice.ddf.registry:type=FederationAdminMBean/regenerateRegistrySources',
+            regenerateSources: function() {
+                var checkboxes = $('.regenerate-source-check');
+                var regIds = [];
+                _.each(checkboxes, function(input){
+                    if(input.checked) {
+                        regIds.push($(input).attr('id'));
+                    }
+                });
+                var mbean = 'org.codice.ddf.registry:type=FederationAdminMBean';
+                var operation = 'regenerateRegistrySources';
+                var modal = this;
+                var data = {
+                    type: 'EXEC',
+                    mbean: mbean,
+                    operation: operation
+                };
+
+                data.arguments = [regIds];
+                data = JSON.stringify(data);
+
+                return $.ajax({
```

Comment:
We might want to restrict the user from leaving or clicking the button again while it's in progress.